packages/core/src/Stores/client-store.js

```diff
@@ -2794,21 +2803,26 @@ export default class ClientStore extends BaseStore {
             target_currency,
         });
 
-        subscription.subscribe(response => {
-            const rates = response.exchange_rates?.rates;
-            const subscription_id = String(response.subscription?.id);
+        subscription.subscribe(
+            response => {
+                const rates = response.exchange_rates?.rates;
+                const subscription_id = String(response.subscription?.id);
 
-            if (rates) {
-                this.addSubscription('exchange_rates', key, undefined, subscription_id);
+                if (rates) {
+                    this.addSubscription('exchange_rates', key, undefined, subscription_id);
 
-                const currentData = { ...this.exchange_rates };
-                if (currentData) {
-                    currentData[base_currency] = { ...currentData[base_currency], ...rates };
-                } else currentData = { [base_currency]: rates };
+                    const currentData = { ...this.exchange_rates };
+                    if (currentData) {
+                        currentData[base_currency] = { ...currentData[base_currency], ...rates };
+                    } else currentData = { [base_currency]: rates };
 
-                this.setExchangeRates(currentData);
+                    this.setExchangeRates(currentData);
+                }
+            },
+            error => {
+                // error handling needed
```

Comment:
What is our error handling?

Reply:
@chrisgodfrey-deriv in client store we just write the error in console like this:

```
// eslint-disable-next-line no-console
console.error('...');
```

On production before my changes we just put this error in `useState` and never use it. I can add console.error if you want or keep it empty as now

Reply:
@chrisgodfrey-deriv Also, we have setError function in common-store, but it will show modal for whole screen, not sure it's a good idea to do this in this case

Reply:
we should still show something to the users, so they understand what is happening as I'm sure not all will view the console errors or Network tab. And this can affect multiple apps, so its better you show a Modal rather than nothing

Reply:
@ameerul-deriv do you want something like this (image attached) for the exchange_rates error??? Or do I need to create another one modal for this? In my opinion the best solution will be adding notifications like this (image attached 2), as example you can see this package (https://www.npmjs.com/package/react-notifications). But anyway it's out of the scope of this card, we don't show the anything for this type of errors. I'm ready to improve our application and make it more client oriented, but it's totally not the task of this PR

<img width="1503" alt="image" src="https://github.com/binary-com/deriv-app/assets/120570511/331c7c14-a7ca-4998-83b4-4035df27b2b5">

<img width="709" alt="image" src="https://github.com/binary-com/deriv-app/assets/120570511/53361542-e62b-46a2-aea1-928217cf4b9e">

Reply:
@ameerul-deriv for that we can use this component:
![image](https://github.com/binary-com/deriv-app/assets/100833613/24120da6-1303-4d2f-8a1b-e545753b9f30)
but it's not just that; we should also prevent the buy/sell ads from rendering if the currency's rate is not available.what do you think?

Reply:
For this card's scope, I believe showing the first modal that we use for everything should be sufficient.  I agree, react-toast can be a improvement to be done in a refined scope.

Reply:
agree @habib-deriv

Reply:
hmmm i get your point but users need to know why they can't be using ZWL. We can't just log the error, we need to handle it properly. This won't completely stop users from using P2P because you need a USD account to be able to use P2P, we only subscribe to ZWL because users try to select ZWL when they want to view any adverts available from there, but theres no Market feed, which causes the issue.

Outside of P2P, I don't necessarily know if we subscribe to these exchange rates that don't have any market feed, but we still need to show either a modal or toast.

If we don't want to stop the entire app since thats what the Modal does, we need to create another one where it doesn't stop everything. It can inform the users please switch your account or ZWL cannot be used, please switch accounts.

Reply:
That's a good idea @ameerul-deriv . We need to have a popup where it doesn't stop everything and forces user to refresh as mentioned by @sergei-deriv. We can look into it further, where we use that in our app. 

However, as discussed, we’ll show the error in the console using console.warn for now and get this production issue released first.